plugins/headlamp-plugin/bin/headlamp-plugin.js

```diff
@@ -316,6 +348,205 @@ function format(packageFolder) {
+      path.join('src', 'headlamp-plugin.d.ts'),
+      'tsconfig.json',
+      'jsconfig.json',
+    ];
+    const templateFolder = path.resolve(__dirname, '..', 'template');
+
+    missingFiles.forEach(pathToCheck => {
+      const from = path.join(templateFolder, pathToCheck);
+      const to = path.join('.', pathToCheck);
+
+      // only copy it if it doesn't exist
+      if (!fs.existsSync(to)) {
+        console.log(`Adding missing file: "${to}"`);
+        fs.copyFileSync(from, to);
+      }
+    });
+  }
+
+  /**
+   * Adds missing config into package.json
+   */
+  function addMissingConfiguration() {
+    const templateFolder = path.resolve(__dirname, '..', 'template');
+    const packageJsonPath = path.join('.', 'package.json');
+    const templatePackageJson = JSON.parse(
+      fs.readFileSync(path.join(templateFolder, 'package.json'), 'utf8')
+    );
+    const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, 'utf8'));
+
+    let changed = false;
+
+    if (packageJson?.scripts?.tsc !== templatePackageJson['scripts']['tsc']) {
+      packageJson['scripts']['tsc'] = templatePackageJson['scripts']['tsc'];
+      changed = true;
+      console.log(`Updated package.json field "scripts.tsc"`);
+    }
+
+    const checkKeys = ['eslintConfig', 'prettier'];
+    checkKeys.forEach(key => {
+      if (JSON.stringify(packageJson[key]) !== JSON.stringify(templatePackageJson[key])) {
+        packageJson[key] = templatePackageJson[key];
+        changed = true;
+        console.log(`Updated package.json field "${key}"`);
+      }
+    });
+
+    if (changed) {
+      fs.writeFileSync(packageJsonPath, JSON.stringify(packageJson, null, '  ') + '\n');
+    }
+  }
+
+  function upgradeHeadlampPlugin(skipPackageUpdates) {
+    if (skipPackageUpdates === true) {
+      console.log('Skipping package updates.');
+      return true;
+    }
+
+    const outDated = getNpmOutdated();
+    if ('@kinvolk/headlamp-plugin' in outDated) {
+      // Upgrade the @kinvolk/headlamp-plugin
+
+      const cmd = 'npm install @kinvolk/headlamp-plugin@latest --save';
+      try {
+        child_process.execSync(cmd, {
+          stdio: 'inherit',
+          encoding: 'utf8',
+        });
+      } catch (e) {
+        console.error(`Problem running ${cmd}`);
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Upgrade a single package in a folder.
+   * @param {string} folder - where the package is.
+   * @param {boolean | undefined} - If true do not update packages.
+   * @returns true if it is successful
+   */
+  function upgradePackage(folder, skipPackageUpdates) {
+    if (!fs.existsSync(path.join(folder, 'package.json'))) {
+      return false;
+    }
+
+    const oldCwd = process.cwd();
+
+    process.chdir(folder);
+    console.log(`Upgrading "${folder}"...`);
+
+    addMissingTemplateFiles();
+    addMissingConfiguration();
+    if (!upgradeHeadlampPlugin(skipPackageUpdates)) {
+      return false;
+    }
```

Comment:
This block led me to think that skipPackageUpdates would be used in the function we're passing it to and if it's false, then the function would also return false; after all, the package would not have been updated. Which would make this whole logic very confusing for the user.
Then I read the upgradeHeadlampPlugin and realized it's returning false just on error.

I think the code would be clearer IMO if you just check the `skipPackageUpdates` here (as `upgradeHeadlampPlugin` is not called from anywhere else).

Reply:
Good point. I moved the skipPackageUpdates check outside that function.